packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,353 @@
+        columnIndex: number,
+        activeColumnDivider: number
+    ): void {
+        this.activeColumnDivider = activeColumnDivider;
+        this.currentTotalDelta = 0;
+        this.initialColumnPixelWidths = [];
+        this.flagActiveColumnDividers(columnIndex);
+        this.setColumnsToFixedSize();
+        this.initialTableWidth = this.table.getBoundingClientRect().width;
+        this.initialTableScrollableWidth = this.initialTableWidth * this.table.tableWidthFactor;
+        this.initialColumnTotalWidth = this.getTotalColumnFixedWidth();
+        this.table.isColumnBeingSized = true;
+        document.addEventListener('mousemove', this.onDividerMouseMove);
+        document.addEventListener('mouseup', this.onDividerMouseUp);
+    }
+
+    public updateTableViewportMinWidth(): void {
+        if (!this.table.$fastController.isConnected) {
+            return;
+        }
+
+        this.table.tableViewportMinWidth = Math.round(
+            (this.table.headerRowActionContainer?.getBoundingClientRect()
+                .width ?? 0)
+                + this.getAllColumnsMinimumWidth()
+                + this.table.virtualizer.headerContainerMarginRight
+        );
+    }
+
+    public handleChange(source: unknown, args: unknown): void {
+        if (
+            source instanceof Virtualizer
+            && args === 'headerContainerMarginRight'
+        ) {
+            this.updateTableViewportMinWidth();
+        }
+    }
+
+    private readonly onDividerMouseMove = (event: Event): void => {
+        const mouseEvent = event as MouseEvent;
+        const deltaX = mouseEvent.movementX > 0
+            ? Math.floor(mouseEvent.movementX)
+            : Math.ceil(mouseEvent.movementX);
+        this.currentTotalDelta += deltaX;
+        for (let i = 0; i < this.table.columns.length; i++) {
+            this.table.columns[i]!.columnInternals.currentPixelWidth = this.initialColumnPixelWidths[i]?.initialPixelWidth;
+        }
+        this.currentTotalDelta = this.pinColumnSizeDelta(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+        this.performCascadeSizeLeft(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+        this.performCascadeSizeRight(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+
+        this.table.tableWidthFactor = Math.max(
+            this.getCurrentTotalTableWidth() / this.initialTableWidth!,
+            1
+        );
+    };
+
+    private readonly onDividerMouseUp = (): void => {
+        document.removeEventListener('mousemove', this.onDividerMouseMove);
+        document.removeEventListener('mouseup', this.onDividerMouseUp);
+        this.unflagActiveColumnDividers();
+        this.updateTableViewportMinWidth();
+        this.resetGridSizedColumns();
+        this.table.isColumnBeingSized = false;
+    };
+
+    private getAllColumnsMinimumWidth(): number {
```

Comment:
Doesn't look like this handles hidden columns.

Reply:
We now are handling hidden columns.